### ISSUE TITLE
Implement plugin as class mixin

### ISSUE THREAD

**Issue by abelsoares** [combine/objection-slugify#1]
First of all, thanks for this plugin, right what i was looking for!

I see that the plugin does not follow the `objection.js` suggestion of implementation as class mixin.  
The problem is that with the current implementation we can to do this:

```
const slugged = Slugify(Model, options);

class Foobar extends Slugged {
...
```
Which is ok, but it becames hard to mixin other plugins, having to do something like this:

```
const slugged = model => Slugify(model, options);

class Foobar extends slugged(otherPlugin(Model)) {
...
```

Would you consider a simple PR to change this plugin to [follow this recommendations](http://vincit.github.io/objection.js/#plugin-development-best-practices)? I have already test it locally and it's an easy fix.

**Comment by calvinl** [comment]
@abelsoares It is implemented as a mixin as far as I can tell (I followed the recommendations when writing it), but I’m definitely willing to see a PR if you’re seeing something I’m not. 

Thanks for contributing!

**Comment by abelsoares** [comment]
I think the implementation should accept options as argument and return a function which accepts `model` as argument. Going to do a PR for you to check out!

**Comment by calvinl** [comment]
Closed via #2.
